### PR TITLE
python311Packages.cliff: 4.2.0 -> 4.3.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 241637` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.optuna</li>
    <li>python310Packages.optuna.dist</li>
    <li>python310Packages.oslo-db</li>
    <li>python310Packages.oslo-db.dist</li>
    <li>python310Packages.subunit2sql</li>
    <li>python310Packages.subunit2sql.dist</li>
    <li>python311Packages.oslo-db</li>
    <li>python311Packages.oslo-db.dist</li>
    <li>python311Packages.subunit2sql</li>
    <li>python311Packages.subunit2sql.dist</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>python311Packages.oslo-log</li>
    <li>python311Packages.oslo-log.dist</li>
    <li>python311Packages.python-manilaclient</li>
    <li>python311Packages.python-manilaclient.dist</li>
    <li>python311Packages.tempest</li>
    <li>python311Packages.tempest.dist</li>
  </ul>
</details>
<details>
  <summary>94 packages built:</summary>
  <ul>
    <li>glanceclient (python310Packages.python-glanceclient)</li>
    <li>glanceclient.dist (python310Packages.python-glanceclient.dist)</li>
    <li>heatclient (python310Packages.python-heatclient)</li>
    <li>heatclient.dist (python310Packages.python-heatclient.dist)</li>
    <li>ironicclient (python310Packages.python-ironicclient)</li>
    <li>ironicclient.dist (python310Packages.python-ironicclient.dist)</li>
    <li>jenkins-job-builder (python310Packages.jenkins-job-builder)</li>
    <li>jenkins-job-builder.dist (python310Packages.jenkins-job-builder.dist)</li>
    <li>manilaclient (python310Packages.python-manilaclient)</li>
    <li>manilaclient.dist (python310Packages.python-manilaclient.dist)</li>
    <li>openstackclient (python310Packages.python-openstackclient)</li>
    <li>openstackclient.dist (python310Packages.python-openstackclient.dist)</li>
    <li>python310Packages.cliff</li>
    <li>python310Packages.cliff.dist</li>
    <li>python310Packages.hacking</li>
    <li>python310Packages.hacking.dist</li>
    <li>python310Packages.keystoneauth1</li>
    <li>python310Packages.keystoneauth1.dist</li>
    <li>python310Packages.openstacksdk</li>
    <li>python310Packages.openstacksdk.dist</li>
    <li>python310Packages.osc-lib</li>
    <li>python310Packages.osc-lib.dist</li>
    <li>python310Packages.oslo-concurrency</li>
    <li>python310Packages.oslo-concurrency.dist</li>
    <li>python310Packages.oslo-config</li>
    <li>python310Packages.oslo-config.dist</li>
    <li>python310Packages.oslo-context</li>
    <li>python310Packages.oslo-context.dist</li>
    <li>python310Packages.oslo-i18n</li>
    <li>python310Packages.oslo-i18n.dist</li>
    <li>python310Packages.oslo-log</li>
    <li>python310Packages.oslo-log.dist</li>
    <li>python310Packages.oslo-serialization</li>
    <li>python310Packages.oslo-serialization.dist</li>
    <li>python310Packages.oslo-utils</li>
    <li>python310Packages.oslo-utils.dist</li>
    <li>python310Packages.python-cinderclient</li>
    <li>python310Packages.python-cinderclient.dist</li>
    <li>python310Packages.python-jenkins</li>
    <li>python310Packages.python-jenkins.dist</li>
    <li>python310Packages.python-keystoneclient</li>
    <li>python310Packages.python-keystoneclient.dist</li>
    <li>python310Packages.python-novaclient</li>
    <li>python310Packages.python-novaclient.dist</li>
    <li>swiftclient (python310Packages.python-swiftclient)</li>
    <li>swiftclient.dist (python310Packages.python-swiftclient.dist)</li>
    <li>python310Packages.stestr</li>
    <li>python310Packages.stestr.dist</li>
    <li>python310Packages.tempest</li>
    <li>python310Packages.tempest.dist</li>
    <li>python311Packages.cliff</li>
    <li>python311Packages.cliff.dist</li>
    <li>python311Packages.hacking</li>
    <li>python311Packages.hacking.dist</li>
    <li>python311Packages.jenkins-job-builder</li>
    <li>python311Packages.jenkins-job-builder.dist</li>
    <li>python311Packages.keystoneauth1</li>
    <li>python311Packages.keystoneauth1.dist</li>
    <li>python311Packages.openstacksdk</li>
    <li>python311Packages.openstacksdk.dist</li>
    <li>python311Packages.osc-lib</li>
    <li>python311Packages.osc-lib.dist</li>
    <li>python311Packages.oslo-concurrency</li>
    <li>python311Packages.oslo-concurrency.dist</li>
    <li>python311Packages.oslo-config</li>
    <li>python311Packages.oslo-config.dist</li>
    <li>python311Packages.oslo-context</li>
    <li>python311Packages.oslo-context.dist</li>
    <li>python311Packages.oslo-i18n</li>
    <li>python311Packages.oslo-i18n.dist</li>
    <li>python311Packages.oslo-serialization</li>
    <li>python311Packages.oslo-serialization.dist</li>
    <li>python311Packages.oslo-utils</li>
    <li>python311Packages.oslo-utils.dist</li>
    <li>python311Packages.python-cinderclient</li>
    <li>python311Packages.python-cinderclient.dist</li>
    <li>python311Packages.python-glanceclient</li>
    <li>python311Packages.python-glanceclient.dist</li>
    <li>python311Packages.python-heatclient</li>
    <li>python311Packages.python-heatclient.dist</li>
    <li>python311Packages.python-ironicclient</li>
    <li>python311Packages.python-ironicclient.dist</li>
    <li>python311Packages.python-jenkins</li>
    <li>python311Packages.python-jenkins.dist</li>
    <li>python311Packages.python-keystoneclient</li>
    <li>python311Packages.python-keystoneclient.dist</li>
    <li>python311Packages.python-novaclient</li>
    <li>python311Packages.python-novaclient.dist</li>
    <li>python311Packages.python-openstackclient</li>
    <li>python311Packages.python-openstackclient.dist</li>
    <li>python311Packages.python-swiftclient</li>
    <li>python311Packages.python-swiftclient.dist</li>
    <li>python311Packages.stestr</li>
    <li>python311Packages.stestr.dist</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
> * python311Packages.oslo-log
> * python311Packages.oslo-log.dist
> * python311Packages.python-manilaclient
> * python311Packages.python-manilaclient.dist
> * python311Packages.tempest
> * python311Packages.tempest.dist

openstack has bad python 3.11 support. I am going to take a look at them in the next days and maybe mark some things broken.

File: pkgs/development/python-modules/cliff/default.nix [hunk 1]
```diff
@@ -1,27 +1,31 @@
...
 , sphinx
-, callPackage
+, stevedore
 }:
 
 buildPythonPackage rec {
   pname = "cliff";
-  version = "4.2.0";
+  version = "4.3.0";
+  format = "setuptools";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/cliff/default.nix:22]: Whut!?

```
pythonOutputDistPhase
Executing pythonOutputDistPhase
The build contains no ./dist directory.
If this project is not setuptools-based, pass
  format = "other";
to buildPythonApplication/buildPythonPackage or another appropriate value as described here:
  https://nixos.org/manual/nixpkgs/stable/#buildpythonpackage-function
```

https://github.com/openstack/cliff/tree/4.3.0

Comment (fabaff) [reply pkgs/development/python-modules/cliff/default.nix:22]: Locally on x86-64 I don't get this error. Not sure whatŝ going on here.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/cliff/default.nix:22]: Could it be actually coming from the tests here https://github.com/fabaff/nixpkgs/blob/cliff-bump/pkgs/development/python-modules/cliff/tests.nix#L10 ?

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 241637` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.optuna</li>
    <li>python310Packages.optuna.dist</li>
    <li>python310Packages.oslo-db</li>
    <li>python310Packages.oslo-db.dist</li>
    <li>python310Packages.subunit2sql</li>
    <li>python310Packages.subunit2sql.dist</li>
    <li>python311Packages.oslo-db</li>
    <li>python311Packages.oslo-db.dist</li>
    <li>python311Packages.subunit2sql</li>
    <li>python311Packages.subunit2sql.dist</li>
  </ul>
</details>
<details>
  <summary>100 packages built:</summary>
  <ul>
    <li>glanceclient (python310Packages.python-glanceclient)</li>
    <li>glanceclient.dist (python310Packages.python-glanceclient.dist)</li>
    <li>heatclient (python310Packages.python-heatclient)</li>
    <li>heatclient.dist (python310Packages.python-heatclient.dist)</li>
    <li>ironicclient (python310Packages.python-ironicclient)</li>
    <li>ironicclient.dist (python310Packages.python-ironicclient.dist)</li>
    <li>jenkins-job-builder (python310Packages.jenkins-job-builder)</li>
    <li>jenkins-job-builder.dist (python310Packages.jenkins-job-builder.dist)</li>
    <li>manilaclient (python310Packages.python-manilaclient)</li>
    <li>manilaclient.dist (python310Packages.python-manilaclient.dist)</li>
    <li>openstackclient (python310Packages.python-openstackclient)</li>
    <li>openstackclient.dist (python310Packages.python-openstackclient.dist)</li>
    <li>python310Packages.cliff</li>
    <li>python310Packages.cliff.dist</li>
    <li>python310Packages.hacking</li>
    <li>python310Packages.hacking.dist</li>
    <li>python310Packages.keystoneauth1</li>
    <li>python310Packages.keystoneauth1.dist</li>
    <li>python310Packages.openstacksdk</li>
    <li>python310Packages.openstacksdk.dist</li>
    <li>python310Packages.osc-lib</li>
    <li>python310Packages.osc-lib.dist</li>
    <li>python310Packages.oslo-concurrency</li>
    <li>python310Packages.oslo-concurrency.dist</li>
    <li>python310Packages.oslo-config</li>
    <li>python310Packages.oslo-config.dist</li>
    <li>python310Packages.oslo-context</li>
    <li>python310Packages.oslo-context.dist</li>
    <li>python310Packages.oslo-i18n</li>
    <li>python310Packages.oslo-i18n.dist</li>
    <li>python310Packages.oslo-log</li>
    <li>python310Packages.oslo-log.dist</li>
    <li>python310Packages.oslo-serialization</li>
    <li>python310Packages.oslo-serialization.dist</li>
    <li>python310Packages.oslo-utils</li>
    <li>python310Packages.oslo-utils.dist</li>
    <li>python310Packages.python-cinderclient</li>
    <li>python310Packages.python-cinderclient.dist</li>
    <li>python310Packages.python-jenkins</li>
    <li>python310Packages.python-jenkins.dist</li>
    <li>python310Packages.python-keystoneclient</li>
    <li>python310Packages.python-keystoneclient.dist</li>
    <li>python310Packages.python-novaclient</li>
    <li>python310Packages.python-novaclient.dist</li>
    <li>swiftclient (python310Packages.python-swiftclient)</li>
    <li>swiftclient.dist (python310Packages.python-swiftclient.dist)</li>
    <li>python310Packages.stestr</li>
    <li>python310Packages.stestr.dist</li>
    <li>python310Packages.tempest</li>
    <li>python310Packages.tempest.dist</li>
    <li>python311Packages.cliff</li>
    <li>python311Packages.cliff.dist</li>
    <li>python311Packages.hacking</li>
    <li>python311Packages.hacking.dist</li>
    <li>python311Packages.jenkins-job-builder</li>
    <li>python311Packages.jenkins-job-builder.dist</li>
    <li>python311Packages.keystoneauth1</li>
    <li>python311Packages.keystoneauth1.dist</li>
    <li>python311Packages.openstacksdk</li>
    <li>python311Packages.openstacksdk.dist</li>
    <li>python311Packages.osc-lib</li>
    <li>python311Packages.osc-lib.dist</li>
    <li>python311Packages.oslo-concurrency</li>
    <li>python311Packages.oslo-concurrency.dist</li>
    <li>python311Packages.oslo-config</li>
    <li>python311Packages.oslo-config.dist</li>
    <li>python311Packages.oslo-context</li>
    <li>python311Packages.oslo-context.dist</li>
    <li>python311Packages.oslo-i18n</li>
    <li>python311Packages.oslo-i18n.dist</li>
    <li>python311Packages.oslo-log</li>
    <li>python311Packages.oslo-log.dist</li>
    <li>python311Packages.oslo-serialization</li>
    <li>python311Packages.oslo-serialization.dist</li>
    <li>python311Packages.oslo-utils</li>
    <li>python311Packages.oslo-utils.dist</li>
    <li>python311Packages.python-cinderclient</li>
    <li>python311Packages.python-cinderclient.dist</li>
    <li>python311Packages.python-glanceclient</li>
    <li>python311Packages.python-glanceclient.dist</li>
    <li>python311Packages.python-heatclient</li>
    <li>python311Packages.python-heatclient.dist</li>
    <li>python311Packages.python-ironicclient</li>
    <li>python311Packages.python-ironicclient.dist</li>
    <li>python311Packages.python-jenkins</li>
    <li>python311Packages.python-jenkins.dist</li>
    <li>python311Packages.python-keystoneclient</li>
    <li>python311Packages.python-keystoneclient.dist</li>
    <li>python311Packages.python-manilaclient</li>
    <li>python311Packages.python-manilaclient.dist</li>
    <li>python311Packages.python-novaclient</li>
    <li>python311Packages.python-novaclient.dist</li>
    <li>python311Packages.python-openstackclient</li>
    <li>python311Packages.python-openstackclient.dist</li>
    <li>python311Packages.python-swiftclient</li>
    <li>python311Packages.python-swiftclient.dist</li>
    <li>python311Packages.stestr</li>
    <li>python311Packages.stestr.dist</li>
    <li>python311Packages.tempest</li>
    <li>python311Packages.tempest.dist</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Update is part of #251878.

**Pull request closed without merging**